SofaKernel/framework/sofa/core/objectmodel/Base.cpp

```diff
@@ -534,6 +532,13 @@ void  Base::parse ( BaseObjectDescription* arg )
         // the same name cannot be extracted from BaseObjectDescription
         if (attrName == std::string("type")) continue;
 
+        if (attrName == std::string("name") && std::string(it.second.c_str()).empty())
+        {
+            msg_warning(getName()) << "Empty name given: Renaming to \"unnamed\""
+                                      " as this can lead to unexpected behaviors.";
+            parseField(attrName, "unnamed");
```

Comment:
I thought there was already a mechanism for that with an increment?

For example if I take a scene like ChainAll, every component has a value for name.

Reply:
That's when the `name` attribute is entirely ommited.
What This PR does is tackle the case where the attribute is forcefully set to an empty string.
But you are right that it would be even better if the fix would use an increment system to stay homoogeneous with the other mechanism, and to avoid getting multiple nodes / components with a same name `unnamed` in the same node.
I'll try to add that feature asap